source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc

```diff
@@ -227,11 +232,16 @@ bool Filter::parseV2Header(char* buf) {
         la6.sin6_family = AF_INET6;
         la6.sin6_port = v6->dst_port;
         safeMemcpy(&(la6.sin6_addr.s6_addr), &(v6->dst_addr));
-
-        proxy_protocol_header_.emplace(WireHeader{
-            hdr_addr_len - PROXY_PROTO_V2_ADDR_LEN_INET6, Network::Address::IpVersion::v6,
-            std::make_shared<Network::Address::Ipv6Instance>(ra6),
-            std::make_shared<Network::Address::Ipv6Instance>(la6)});
+        StatusOr<Network::Address::InstanceConstSharedPtr> error_or_address_la =
+            Network::Address::InstanceFactory::createInstancePtr<Network::Address::Ipv6Instance>(
+                la6);
+        StatusOr<Network::Address::InstanceConstSharedPtr> error_or_address_ra =
+            Network::Address::InstanceFactory::createInstancePtr<Network::Address::Ipv6Instance>(
+                ra6);
+        ASSERT(error_or_address_la.ok() && error_or_address_ra.ok());
```

Comment:
RELEASE_ASSERT rather than ASSERT. Imagine you are debugging a production crash. If you just ASSERT here you will probably have a seg-fault on the next line. So you want to have the assert compiled into release builds, and have that provide as much context as possible, in particular the `.message()` result from the status.

Or use the "OrDie" variants of the helper methods, if that's what's really needed.